### PR TITLE
Add an interface of disabling FC padding

### PR DESCRIPTION
**Pull request by GaoWei8**
Add an interface of disabling FC padding.
Padding is applied by default.

ERNIE performance：

|Padding | Disable padding |
|---|---|
| 26.6749 ms| 46.3951 ms |

BERT：

|Padding | Disable padding |
|---|---|
|  fps: 111.36| fps: 115.81 |

### CODE REVIEW

**Comment by Xreki** [comment]
你在paddle/fluid目录下面执行`grep -r use_fc_padding *`看下：
```bash
$ grep -r use_fc_padding *
framework/ir/fc_fuse_pass.cc:    bool use_fc_padding =
framework/ir/fc_fuse_pass.cc:        Has("use_fc_padding") ? Get<bool>("use_fc_padding") : true;
framework/ir/fc_fuse_pass.cc:    LOG(INFO) << "use_gpu:" << use_gpu << ", use_fc_padding:" << use_fc_padding;
framework/ir/fc_fuse_pass.cc:    if (!use_gpu && use_fc_padding) {
inference/analysis/ir_pass_manager.cc:  bool use_fc_padding = !HasPass("fc_mkldnn_pass");
inference/analysis/ir_pass_manager.cc:    fc_fuse_pass->Set<bool>("use_fc_padding", new bool(use_fc_padding));
```

**Comment by GaoWei8** [comment]
> 你在paddle/fluid目录下面执行`grep -r use_fc_padding *`看下：
> 
> ```shell
> $ grep -r use_fc_padding *
> framework/ir/fc_fuse_pass.cc:    bool use_fc_padding =
> framework/ir/fc_fuse_pass.cc:        Has("use_fc_padding") ? Get<bool>("use_fc_padding") : true;
> framework/ir/fc_fuse_pass.cc:    LOG(INFO) << "use_gpu:" << use_gpu << ", use_fc_padding:" << use_fc_padding;
> framework/ir/fc_fuse_pass.cc:    if (!use_gpu && use_fc_padding) {
> inference/analysis/ir_pass_manager.cc:  bool use_fc_padding = !HasPass("fc_mkldnn_pass");
> inference/analysis/ir_pass_manager.cc:    fc_fuse_pass->Set<bool>("use_fc_padding", new bool(use_fc_padding));
> ```

```
$ grep -r use_fc_padding *
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
framework/ir/fc_fuse_pass.cc:      bool use_fc_padding =
framework/ir/fc_fuse_pass.cc:          Has("use_fc_padding") ? Get<bool>("use_fc_padding") : true;
framework/ir/fc_fuse_pass.cc:      if (!use_gpu && use_fc_padding) {
inference/analysis/ir_pass_manager.cc:  bool use_fc_padding = !HasPass("fc_mkldnn_pass");
inference/analysis/ir_pass_manager.cc:    fc_fuse_pass->Set<bool>("use_fc_padding", new bool(use_fc_padding));
```

**Comment by ddokupil** [comment]
It looks like we're back with performance from release 1.6.

**Comment by ddokupil** [comment]
We have 133.741FPS on this vs 127.834FPS on develop. Can we please let this fix in?

**Comment by GaoWei8** [comment]
> We have 133.741FPS on this vs 127.834FPS on develop. Can we please let this fix in?

Sure. This PR is waiting for review. Thanks for reminding.

File: paddle/fluid/framework/ir/fc_fuse_pass.cc [hunk 1]
```diff
@@ -90,32 +90,35 @@ int FCFusePass::ApplyFCPattern(Graph* graph, bool with_relu) const {
...
+      auto* scope = param_scope();
+      auto* weight = scope->FindVar(w->Name())->GetMutable<LoDTensor>();
+      auto* weight_data = weight->data<float>();
+      auto weight_dims = weight->dims();
+      int weight_num = product(weight_dims);
+      int w_h = weight_dims[0];
+      int w_w = weight_dims[1];
+      bool use_gpu = Has("use_gpu") ? Get<bool>("use_gpu") : false;
+      bool use_fc_padding =
+          Has("use_fc_padding") ? Get<bool>("use_fc_padding") : true;
```
Comment (Xreki) [inline paddle/fluid/framework/ir/fc_fuse_pass.cc]: 不需要同时存在`use_fc_padding`和`use_padding`这样两个属性，也不需要两层if判断。从代表的含义上来看，`use_fc_padding`更准确一些。

File: paddle/fluid/inference/analysis/argument.h [hunk 2]
```diff
@@ -173,6 +173,7 @@ struct Argument {
 
   // Passed from config.
   DECL_ARGUMENT_FIELD(use_gpu, UseGPU, bool);
+  DECL_ARGUMENT_FIELD(use_padding, UsePadding, bool);
```
Comment (Xreki) [inline paddle/fluid/inference/analysis/argument.h]: 这里同样，用`use_fc_padding`更准确一些。这里要增加预测接口，看看是不是需要请预测的同学也来review下。

File: paddle/fluid/framework/ir/fc_fuse_pass.cc (hunk 1 above)
Comment (GaoWei8) [reply paddle/fluid/framework/ir/fc_fuse_pass.cc]: Done

File: paddle/fluid/inference/analysis/argument.h (hunk 2 above)
Comment (GaoWei8) [reply paddle/fluid/inference/analysis/argument.h]: Done

**Review by Xreki — APPROVED** [review]
LGTM

**Review by Shixiaowei02 — APPROVED** [review]

**Pull request merged**